src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +386,55 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
 		}
 		ioOperations.deleteInventory(medicalInventory);
 	}
+
+	/**
+	 * Confirm the Inventory rows of inventory.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
+	 * @throws OHDataValidationException
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void confirmMedicalInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		// validate the inventory
+		this.validateMedicalInventoryRow(inventory, inventoryRowSearchList);
+		// create Movements
+		String dischargeCode = inventory.getDischargeType();
+		String chargeCode = inventory.getChargeType();
+		Integer supplierId = inventory.getSupplier();
+		String wardCode = inventory.getDestination();
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(chargeCode);
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(dischargeCode);
+		Supplier supplier = supplierManager.getByID(supplierId);
+		Ward ward = wardManager.findWard(wardCode);
+		LocalDateTime now = TimeTools.getNow();
+		for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+			MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+			StringBuilder referenceBuilder = new StringBuilder();
+			referenceBuilder.append(inventory.getInventoryReference())
+			                .append("-")
+			                .append(now);
+			double theoQty = medicalInventoryRow.getTheoreticQty();
+			double realQty = medicalInventoryRow.getRealQty();
+			Double ajustQty = theoQty - realQty;
+			Medical medical = medicalInventoryRow.getMedical();
+			String lotCode = medicalInventoryRow.getLot().getCode();
+			Lot currentLot = movStockInsertingManager.getLot(lotCode);
+			if (realQty > theoQty) { // charge movement when realQty > theoQty
+				referenceBuilder.append("-charge");
+				String reference = referenceBuilder.toString();
+				Movement movement = new Movement(medical, chargeType, null, currentLot, now, -(ajustQty.intValue()), supplier, reference);
+				List<Movement> chargeMovement = new ArrayList<>();
+				chargeMovement.add(movement);
+				chargeMovement = movStockInsertingManager.newMultipleChargingMovements(chargeMovement, reference);
```

Comment:
you are inserting only one movement... this should be likely called after the iteration cycle, and only if `chargeMovement` is not empty
```suggestion
				chargeMovement = movStockInsertingManager.newMultipleChargingMovements(chargeMovement, reference);
```

Reply:
Moreover there is a problem with AUTOMATICLOT_IN=yes: before the inventory feature, the charging movements were done generating automatically a new lot code and asking the user only the expiring date. 
Now, instead, we want to insert a charge movement where the lot is specified (because already created/generated).
We didn't take this into account when creating new lots during the inventory... let's think how to solve this problem...

Option 1)
We don't create lots during the inventory and leave the system to do it during confirmation and according with the general settings, using the data in the inventory row. In this case we can also avoid to delete new lots when deleting an inventory or an inventoryrow with "N"

Option 2)
We change the way a new charge movement is created by taking into account that the lot might be provided even if the AUTOMATICLOT_IN=yes. 
> For instance the check in `MedicalStockIoOperations.newMovement(Movement)` at the moment is `if (isAutomaticLotMode() || "".equals(lotCode))`
> Maybe is enough to change it to `if (isAutomaticLotMode() && "".equals(lotCode))`, but it needs testing

In the case 2) we have to pay attention to not brake previous behaviour.

Reply:
pushed a change for Option 2)